ledger-service/jwt/src/main/scala/com/digitalasset/jwt/JwtVerifier.scala

```diff
@@ -44,12 +52,45 @@ object HMAC256Verifier extends StrictLogging {
   def apply(secret: String): Error \/ JwtVerifier =
     \/.fromTryCatchNonFatal {
       logger.warn(
-        "HMAC256 JWT Validator is NOT recommended for production env, please use RSA256 (WIP)!!!")
+        "HMAC256 JWT Validator is NOT recommended for production environments, please use RSA256!!!")
 
       val algorithm = Algorithm.HMAC256(secret)
       val verifier = JWT.require(algorithm).build()
       new JwtVerifier(verifier)
     }.leftMap(e => Error('HMAC256, e.getMessage))
 }
 
-// TODO(Leo) RSA256 validator
+// RSA256 validator factory
+object RSA256Verifier extends StrictLogging {
+  def apply(publicKey: RSAPublicKey): Error \/ JwtVerifier =
+    \/.fromTryCatchNonFatal {
+
+      val algorithm = Algorithm.RSA256(publicKey, null)
+      val verifier = JWT.require(algorithm).build()
+      new JwtVerifier(verifier)
+    }.leftMap(e => Error('RSA256, e.getMessage))
+
+  def apply(keyProvider: RSAKeyProvider): Error \/ JwtVerifier =
+    \/.fromTryCatchNonFatal {
+
+      val algorithm = Algorithm.RSA256(keyProvider)
+      val verifier = JWT.require(algorithm).build()
```

Comment:
right here before calling `build`, I thought you need to ask it to check for expiration with `acceptExpiresAt` or something else... might be wrong. Looked at the docs more than 4 months ago.

Reply:
As per our conversation on slack, we explicitly check expiration ourselves in gRPC requests and their streaming responses.